sky/data/storage.py

```diff
@@ -1871,6 +1904,542 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+                        authorization_client = data_utils.create_az_client(
+                            'authorization')
+                        graph_client = data_utils.create_az_client('graph')
+                        async def get_object_id():
+                            import logging
+                            original_level = logging.getLogger("httpx").getEffectiveLevel()
+                            # silencing the INFO level response log from httpx request
+                            logging.getLogger("httpx").setLevel(logging.WARNING)
+                            user = await graph_client.users.with_url(
+                                'https://graph.microsoft.com/v1.0/me').get()
+                            logging.getLogger("httpx").setLevel(original_level)
+                            object_id = str(user.additional_data['id'])
+                            return object_id
+                        object_id = asyncio.run(get_object_id())
+                        # Defintion ID of Storage Blob Data Owner role.
+                        # Reference: https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-blob-data-owner # pylint: disable=line-too-long
+                        storage_blob_data_owner_role_id = 'b7e6dc6d-f1e8-4753-8033-0f276bb0955b'
+                        role_definition_id = ('/subscriptions'
+                                            f'/{azure.get_subscription_id()}'
+                                            '/providers/Microsoft.Authorization'
+                                            '/roleDefinitions'
+                                            f'/{storage_blob_data_owner_role_id}')
+                        authorization_client.role_assignments.create(
+                                scope=creation_response.id,
+                                role_assignment_name=uuid.uuid4(),
+                                parameters={
+                                    "properties": {
+                                        "principalId": f'{object_id}',
+                                        "principalType": "User",
+                                        "roleDefinitionId": role_definition_id,
+                                    }
+                                },
+                            )
+                    except azure.exceptions().ResourceExistsError as e:
+                        error_message = e.message
+                        if 'StorageAccountAlreadyTaken' in error_message:
+                            with ux_utils.print_exception_no_traceback():
+                                raise exceptions.StorageBucketCreateError(
+                                    'The storage account name '
+                                    f'{self.storage_account_name!r} is already taken. '
+                                    'Please try with another name.')
```

Comment:
`L2069` with `Code: ResourceNotFound' in e.message:` checks if the storage account name exists under current user's account. And this error happens if storage account name attempted to be created already exists globally. Added a comment to avoid confusion: [fb61be4](https://github.com/skypilot-org/skypilot/pull/3032/commits/fb61be40a163f97f0eceb8e6291be57817cca708)

Reply:
Got it! Makes sense.